torch/csrc/distributed/rpc/message.cpp

```diff
@@ -120,6 +130,48 @@ void Message::setId(int64_t id) {
   id_ = id;
 }
 
+at::IValue Message::toIValueTuple() const {
+  // Message has a std::vector<char> payload_ that is represented via a
+  // string, list of tensors, integer message type, and int64 id.
+  std::string payload(payload_.begin(), payload_.end());
```

Comment:
This might be a pretty expensive copy. Instead of having methods to convert to and from IValue, can we change the internal representation of message to use IValue directly? We can add a method to Message to get this IValue and use it in our Future class.

This might be a much bigger change, but it seems like the right way to do this since most of our `toMessageImpl()` serializers create IValues anyways and then use jit pickle to create the payload.

Reply:
@pritamdamania87 Good point, that sounds good. For changing the internal representation to be IValue, do you mean that all the class members such as payload, tensors, message type, and ID should be IValue types? Wouldn't creating the payload IValue as a string in `toMessageImpl()` still result in the copy that we are trying to avoid here?